### DISCUSSION TITLE
Authentik and Apache Guacamole

### DISCUSSION THREAD

**Discussion by netxchanges** [goauthentik/authentik#3336] [Q&A]
Hi all, apologies in advance for the noobish question but am struggling a bit with this integration. 

Intagration documentation is quite basic and looking on google ends up half of the time with stuff like "The real authentic guacamole recipe" ^^'.

So my Authentic installation is behing an Nginx Proxy Manager and perfectly working with Gitea (behind the same NPM)

Maybe I missed something from the integration part but when I have setup guacamole without Authentik integration, things where running fine (Just for the record my NPM config)
For guacamole:
![image](https://user-images.githubusercontent.com/3584068/182032674-6f9bdc1f-37ce-4896-86dc-88b6ac6a5f73.png)
![image](https://user-images.githubusercontent.com/3584068/182032697-11cb6c06-b5b7-4bc7-9143-d78528214996.png)
![image](https://user-images.githubusercontent.com/3584068/182032724-4e92f8ed-efd9-4c36-832a-28b57a251851.png)

For Authentik:
![image](https://user-images.githubusercontent.com/3584068/182032761-4819c997-f1c2-4fca-9335-ae74f5ff5abd.png)
![image](https://user-images.githubusercontent.com/3584068/182032774-1f9d0243-852c-4291-b66e-8dcd302b7117.png)

Now following the instructions on the integration page : 
- Update the docker env variables 
![image](https://user-images.githubusercontent.com/3584068/182031682-1027a116-3634-4858-bec7-5248282a279b.png)

- Defined the application in Authentik  
![image](https://user-images.githubusercontent.com/3584068/182031723-7721aa6e-5d88-4cf4-b359-708d28729058.png)

- Defined the provider in Authentik
![image](https://user-images.githubusercontent.com/3584068/182032871-d231b688-f4e4-4e59-91ba-8543ee0f761f.png)

As soon as I go to guacamole URI, it redirects well to Authentik and once logged on it returns to guacamole, who then returns to authentik, who returns..... endless loop.

Read somewhere self signed certificate in Authentik could cause issues so my NPM and Authentik have the same wildcard certificate generated at Cloudflare.

After few hours trying and nothing really changing, getting out of ideas

**Comment by incoggnito** [comment]
You need to set the correct provider settings.
Check that there is an Self-signed Certificate and the **Token validity** is below 300 minutes.
I've not changed the ngnix special settings and just used https://vdi.mysite.net/guacamole/ as redirect uri.

I currently struggle to set an authentik admin user.
Maybe you can help me ;)
